apps/hvn/src/app/[locale]/announcements/[slug]/page.tsx

```diff
@@ -0,0 +1,65 @@
+import configPromise from '@payload-config'
+import { draftMode } from 'next/headers'
+import { notFound } from 'next/navigation'
+import { getPayload, TypedLocale } from 'payload'
+
+import { AnnouncementDetail } from '@/components/announcement/AnnouncementDetail'
+import { Announcement } from '@/payload-types'
+import config from '@/payload.config'
+
+export async function generateStaticParams() {
+  const payload = await getPayload({ config: configPromise })
+
+  const pages = await payload.find({
+    collection: 'announcements',
+    draft: false,
+    limit: 1000,
+    overrideAccess: false,
+    pagination: false,
```

Comment:
```suggestion
    pagination: false,
    where: {
      _status: {
        equals: 'published',
      }
    }
```

there is no need to add unpublished slugs. because in component you are already filter them out.